### DISCUSSION TITLE
I get INVALID_CLIENT: Invalid redirect URI when opening the URI given from AuthorizationCodeUriExample

### DISCUSSION THREAD

**Discussion by IGprojects** [spotify-web-api-java/spotify-web-api-java#327] [Q&A] [answered]
I have changed the clientId and CLientSecret variables with the ones I created in SpotifyforDevelopers. The problem is that when I run the file and try the URIs, none of them work for me, they give me the following error: INVALID_CLIENT: Invalid redirect URI and I don't know what I'm doing wrong because I have put the URI that it gives me in Edit Setting->RedirectsURIs and it still doesn't work, the code I have in the file is the following:

package com.example.spotifybackupprojectuf2;

import se.michaelthelin.spotify.SpotifyApi;
import se.michaelthelin.spotify.SpotifyHttpManager;
import se.michaelthelin.spotify.requests.authorization.authorization_code.AuthorizationCodeUriRequest;

import java.net.URI;
import java.util.concurrent.CancellationException;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionException;

public class AuthorizationCodeUri{
    private static final String clientId = "...";
    private static final String clientSecret = "...";
    private static final URI redirectUri = SpotifyHttpManager.makeUri("https://example.com/spotify-redirect");

    private static final SpotifyApi spotifyApi = new SpotifyApi.Builder()
            .setClientId(clientId)
            .setClientSecret(clientSecret)
            .setRedirectUri(redirectUri)
            .build();
    private static final AuthorizationCodeUriRequest authorizationCodeUriRequest = spotifyApi.authorizationCodeUri()
//          .state("x4xkmn9pu3j6ukrs8n")
//          .scope("user-read-birthdate,user-read-email")
//          .show_dialog(true)
            .build();

    public static void authorizationCodeUri_Sync() {
        final URI uri = authorizationCodeUriRequest.execute();

        System.out.println("URI: " + uri.toString());
    }

    public static void authorizationCodeUri_Async() {
        try {
            final CompletableFuture<URI> uriFuture = authorizationCodeUriRequest.executeAsync();

            // Thread free to do other tasks...

            // Example Only. Never block in production code.
            final URI uri = uriFuture.join();

            System.out.println("URI: " + uri.toString());
        } catch (CompletionException e) {
            System.out.println("Error: " + e.getCause().getMessage());
        } catch (CancellationException e) {
            System.out.println("Async operation cancelled.");
        }
    }

    public static void main(String[] args) {
        authorizationCodeUri_Sync();
        authorizationCodeUri_Async();
    }
}

**Comment by dargmuesli** [comment, marked as answer]
Hey @IGprojects, make sure to format all lines of code properly. So you're sure you've entered `https://example.com/spotify-redirect` as redirect uri under https://developer.spotify.com/dashboard/? Did you save your changes there?

You get a well formed response from the api indicating a mistake about the redirect uri and in all cases I witnessed the api was right. Doublecheck please, or contact people in closer relation to Spotify as on here you'll only find volunteers. Try StackOverflow as described in the [CONTRIBUTING.md](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/develop/CONTRIBUTING.md#your-questions).

**Reply by IGprojects** [reply]
Yeah i solve that but is not giving me the acces token it redirects me ones i have accept the permision to get info of my spotify profile goes to a page that put only for Example Domain and it doesn't give me the acces token or it's supposed to do it in other way? thanks for the help

**Reply by dargmuesli** [reply]
The access token will be part of the url's query string once you land on `example.com`. Please read the [authorization guide](https://developer.spotify.com/documentation/general/guides/authorization/).
